network-api/networkapi/wagtailpages/factory/libraries/research_hub/__init__.py

```diff
@@ -46,20 +46,22 @@ def generate(seed):
     print("Generating research hub")
 
     # Only one landing page can exist
-    rcc_landing_page = wagtailpage_models.ResearchLandingPage.objects.first()
-    if not rcc_landing_page:
-        rcc_landing_page = landing_page_factory.ResearchLandingPageFactory.create(parent=home_page)
+    research_landing_page = wagtailpage_models.ResearchLandingPage.objects.first()
+    if not research_landing_page:
+        research_landing_page = landing_page_factory.ResearchLandingPageFactory.create(
+            parent=home_page, aside_cta__0="cta", aside_cta__1="cta"
```

Comment:
Also curious if the `"cta"` has any relationship with [class CTABase](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/network-api/networkapi/wagtailpages/pagemodels/base.py#L501-L546) ?

Reply:
This bit is confusing because we are using the factory to create 2 default `aside_cta`s. Check out [`wagtail-factories` for examples](https://github.com/wagtail/wagtail-factories). Basically, to create one, you put the field that you want `aside_cta`, followed by the index of the `StreamField` (in this case 0 and 1, the first two), and then set the value that you want. In our case, we want a default `"cta"`. The string `"cta"` is the name of the block that we want to use on this field, which comes from the definition of the `StreamField` on the Landing Page:

```python
class RCCLandingPage(BaseLandingPage):
    ...
    aside_cta = fields.StreamField(
        block_types=(("cta", customblocks.CTAAsideBlock()),),
        null=True,
        blank=True,
        use_json_field=True,
        max_num=2,
    )
```

If we had something like:

```python
class RCCLandingPage(BaseLandingPage):
    ...
    aside_cta = fields.StreamField(
        block_types=(("cta", customblocks.CTAAsideBlock()),("rich_text", RichTextBlock())),
        null=True,
        blank=True,
        use_json_field=True,
        max_num=2,
    )
```
then we could create one `aside_cta` entry using a default `"cta"` block (`aside_cta__0="cta"`) and another one using a "rich_text" block (`aside_cta__1="rich_text"`), for example. Does that make sense?

Here `"cta"` is just an id/name to identify which block type we want. We could change that to `"my_custom_cta_block"` on both places and it would still work.

 I'm sorry that was confusing :)

Reply:
Thank you for this thorough explanation!